src/DiagnosticTools.jl

```diff
@@ -99,3 +106,30 @@ function dump_diagnostics_shallow_water!(h_tmp, w_tmp,
              step, mass_i, vort_i, kin_i, pot_i, kin_i+pot_i, pow_i)
   end
 end
+
+function dump_diagnostics_thermal_shallow_water!(h_tmp, w_tmp,
+                                         model, dΩ, dω, S, L2MM, H1MM,
+                                         h, u, E, w, ϕ, F, eF, step, dt,
+                                         output_file, dump_on_screen)
+
+  mass_i    = compute_total_mass!(h_tmp, L2MM, get_free_dof_values(h))
```

Comment:
I would wrap around lines 115-119 in a function, say, `compute_diagnostics_shallow_water!`, that can be called both from `dump_diagnostics_thermal_shallow_water!` and `dump_diagnostics_shallow_water!`